sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -187,6 +191,110 @@ public virtual async Task<Response<LogsBatchQueryResult>> QueryBatchAsync(LogsBa
+                        $"(bool(null), datetime(null), dynamic(null), guid(null), int(null), long(null), real(null), double(null), time(null))"),
+
+                    // Boolean
+                    true => "true",
+                    false => "false",
+
+                    // Numeric
+                    sbyte x => $"int({x.ToString(CultureInfo.InvariantCulture)})",
+                    byte x => $"int({x.ToString(CultureInfo.InvariantCulture)})",
+                    short x => $"int({x.ToString(CultureInfo.InvariantCulture)})",
+                    ushort x => $"int({x.ToString(CultureInfo.InvariantCulture)})",
+                    int x => $"int({x.ToString(CultureInfo.InvariantCulture)})",
+                    uint x => $"int({x.ToString(CultureInfo.InvariantCulture)})",
+
+                    float x => $"real({x.ToString(CultureInfo.InvariantCulture)})",
+                    double x => $"real({x.ToString(CultureInfo.InvariantCulture)})",
+
+                    // Int64
+                    long x => $"long({x.ToString(CultureInfo.InvariantCulture)})",
+                    ulong x => $"long({x.ToString(CultureInfo.InvariantCulture)})",
+
+                    decimal x => $"decimal({x.ToString(CultureInfo.InvariantCulture)})",
+
+                    // Guid
+                    Guid x => $"guid({x.ToString("D", CultureInfo.InvariantCulture)})",
+
+                    // Dates as 8601 with a time zone
+                    DateTimeOffset x => $"datetime({x.UtcDateTime.ToString("O", CultureInfo.InvariantCulture)})",
+                    DateTime x => $"datetime({x.ToUniversalTime().ToString("O", CultureInfo.InvariantCulture)})",
+                    TimeSpan x => $"time({x.ToString("c", CultureInfo.InvariantCulture)})",
+
+                    // Text
+                    string x => EscapeStringValue(x),
+                    char x => EscapeStringValue(x),
+
+                    // Everything else
+                    object x => throw new ArgumentException(
+                        $"Unable to convert argument {i} from type {x.GetType()} to an Kusto literal.")
+                };
+            }
+
+            return string.Format(CultureInfo.InvariantCulture, filter.Format, args);
+        }
+
+        private static string EscapeStringValue(string s)
+        {
+            StringBuilder escaped = new();
+            escaped.Append('"');
+
+            foreach (char c in s)
+            {
+                switch (c)
+                {
+                    case '"':
+                        escaped.Append("\\\"");
+                        break;
+                    case '\\':
+                        escaped.Append("\\\\");
+                        break;
+                    case '\r':
+                        escaped.Append("\\r");
+                        break;
+                    case '\n':
+                        escaped.Append("\\n");
```

Comment:
Does this work with [multi-line string literals](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/scalar-data-types/string#multi-line-string-literals)?

Reply:
We don't generate multi-line literals, only the normal ones `"..."` with newlines escaped.